docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,307 @@
+
+[[development-default-editor]]
+==== `default` editor controller
+The default editor controller can receive `optionsTemplate` or `optionsTabs` parameter which can be either an angular template or react component. React component will receive two params: scope, which is an angular scope and stageEditorParams which will update the editor with parameters set in react component.
+
+["source","js"]
+-----------
+{
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    editorController: 'default',
+    editorConfig: {
+       optionsTemplate: '<my-custom-options-directive></my-custom-options-directive>' // or
+       optionsTabs: [
+           { title: 'tab 1', template: '<div>....</div> },
+           { title: 'tab 2', template: '<my-custom-options-directive></my-custom-options-directive>' }
+       ]
+    }
+  }
+-----------
+
+[[development-custom-editor]]
+==== custom editor controller
+You can create a custom editor controller. To do so pass an Editor object (the same format as VisController class). 
+You can make your controller take extra configuration which is passed to the editorConfig property.
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+class MyEditorController {
+    constructor(el, vis) {
+      this.el = el;
+      this.vis = vis;
+   }
+   render(visData) {
+      return new Promise(resolve => {
+         ...
+         resolve('done rendering');
+      }),
+   destroy() {
+      console.log('destroying');
+   }
+}
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    editorController: MyEditorController,
+    editorConfig: { my: 'custom config' }
+  });
+}
+-----------
+
+[[development-visualization-request-handlers]]
+=== Visualization Request Handlers
+By default visualizations will use courier request handler. They can also choose to use any of the other provided 
+request handlers. Its also possible to define your own request handler 
+(which you can then register to be used by other visualizations)
+
+[[development-default-request-handler]]
+==== courier request handler
+'courier' is the default request handler which works with out 'default' side bar editor.
+
+[[development-none-request-handler]]
+==== `none` request handler
+Using 'none' as your request handles means your visualization does not require search.
+
+[[development-custom-request-handler]]
+==== custom request handler
```

Comment:
I think we would need a bit more explanation about what this function does, or make it clear in the docs, that this is kind of an internal thing, and you have to lookup in code how that function works (maybe with link to current implementation).